code/dce-downloader/node/dce.js

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env node
+
+const options = require('commander');
+const zlib = require('zlib');
+const dce = require('./dce-common.js');
+const errors = require('request-promise/errors');
+const util = require('util');
+
+options
+    .version('1.0.0')
+    .option('-p, --path [DCE_file]', 'DCE file path')
+    .option('-c, --config [configFile]', 'path to configuration (default is ../config.json)', '../config.json')
+    .option('-d, --dest [directory]', 'destination folder to store downloaded data')
+    .option('-e, --env <environment>', 'environment of DCE service (must be present in config file)')
+    .parse(process.argv);
+
+const config = dce.readConfig(options.config);
+
+const destination = options.dest;
+const path = options.path;
+
+const environment = dce.getEnvironment(config, options.env) || process.exit(1);
+
+dce.doHttpGet(environment.url, environment.passkey, environment.secret, path)
+    .then(function(response) {
```

Comment:
Node 6+ supports arrow function syntax. This is usually used with easily creating callbacks functions.

```javascript
asyncOperation().then((response) => { ... });
```